core/jvm/src/SegmentPool.kt

```diff
@@ -62,6 +109,14 @@ internal actual object SegmentPool {
     private val HASH_BUCKET_COUNT =
         Integer.highestOneBit(Runtime.getRuntime().availableProcessors() * 2 - 1)
 
+    private val HASH_BUCKET_COUNT_L2 = (HASH_BUCKET_COUNT / 2).coerceAtLeast(1)
+
+    private val SECOND_LEVEL_POOL_TOTAL_SIZE =
+        System.getProperty("kotlinx.io.pool.size.bytes", "0").toInt().coerceAtLeast(0)
```

Comment:
Discussed with @e5l that it makes sense to bump up the default size up to several megs (probably, 4Mb).

Not sure if it should be done on Android.

Reply:
What really puzzles me here is how it affects the end users.
Because it seems like the characteristics of the IO (which is *the* way to do a lot of stuff) will depend on whether or not Ktor is used in the project.

I don't mind it as a palliative intermediate solution as the ground for further benchmarking, but definitely not something to keep as as

Reply:
I checked how `BufferReadWriteByteArray` performs depending on the presence of the second-level pool and the working set size.

If WSS is within L1-pool, there's no difference.
If WSS doesn't fit into L1-pool, it might be beneficial not to use L2-pool (i.e. it's faster w/o L2-pool) until some threshold size, but then L2-pool boosts the performance.
But it's hard to judge by looking at synthetic benchmarks only, as they provide zero to no evidence regarding GC-related effects on overall performance.